### PR TITLE
Better error message when repo does not exist

### PR DESCRIPTION
**Pull request by d2lam**
Better error message

Relates to https://github.com/screwdriver-cd/screwdriver/issues/297

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -180,7 +180,30 @@ describe('index', () => {
                 .then(() => assert.fail('Should not get here'))
                 .catch((error) => {
                     assert.calledWith(requestMock, expectedOptions);
-                    assert.match(error.message, 'STATUS CODE 404');
+                    assert.match(error.message, 'Cannot find repository');
```
Comment (tkyi) [inline test/index.test.js]: Should error.message be `Cannot find repository https://batman@bitbucket.org/batman/test.git#mynewbranch`?

Comment (d2lam) [inline test/index.test.js]: It is but this is doing `assert.match` not `equal`

Comment (tkyi) [inline test/index.test.js]: Ohh I see 👍

File: test/index.test.js [hunk 2]
```diff
@@ -180,7 +180,30 @@ describe('index', () => {
...
+                });
+        });
+
+        it('rejects if status code is not 200 & 404', () => {
+            fakeResponse = {
+                statusCode: 500,
+                body: {
+                    error: {
+                        message: 'INternal Server Error'
```
Comment (FenrirUnbound) [inline test/index.test.js]: Spelling

File: test/index.test.js [hunk 3]
```diff
@@ -180,7 +180,30 @@ describe('index', () => {
...
+                }
+            };
+
+            requestMock.yieldsAsync(null, fakeResponse, fakeResponse.body);
+
+            return scm.parseUrl({
+                checkoutUrl: 'https://batman@bitbucket.org/batman/test.git#mynewbranch',
+                token: 'myAccessToken'
+            })
+                .then(() => assert.fail('Should not get here'))
```
Comment (FenrirUnbound) [inline test/index.test.js]: Combine  the `.catch` clause with this `.then`

File: index.js [hunk 4]
```diff
@@ -101,6 +101,9 @@ class BitbucketScm extends Scm {
 
         return this.breaker.runCommand(options)
             .then((response) => {
+                if (response.statusCode === 404) {
```
Comment (FenrirUnbound) [inline index.js]: For Github, it returns a 404 regardless if the the repository is private or the repository is missing.

What status code does Bitbucket return when it's a private repository?

Comment (d2lam) [inline index.js]: Same. 404.

File: test/index.test.js (hunk 3 above)
Comment (d2lam) [inline test/index.test.js]: I think we went over this. Both ways are fine but some people commented that the other way is hard to read so I switched to this one. Fine with me either way.

Comment (FenrirUnbound) [inline test/index.test.js]: It's confusing this way too. If the `assert.fail` is hit, it'll be caught by the next `.catch` in the chain. If you're lucky, you'll have an error that says "'Should not get here' does not equal 'STATUS CODE 500'". And that's assuming mocha unrolls your unit test error message

**Pull request merged**